### PR TITLE
UPSTREAM: EmptyDir volumes for non-root 2/2

### PR DESCRIPTION
**Pull request by pmorie**
For https://github.com/GoogleCloudPlatform/kubernetes/pull/9384

@pweil- @smarterclayton

### CODE REVIEW

**Comment by pmorie** [comment]
[test]

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir/empty_dir.go [hunk 1]
```diff
@@ -207,6 +203,47 @@ func (ed *emptyDir) setupTmpfs(dir string) error {
...
+			return err
+		}
+
+		fileinfo, err = os.Lstat(dir)
+		if err != nil {
+			return err
+		}
+
+		if fileinfo.Mode().Perm() != perm.Perm() {
+			glog.Errorf("Unexpected mode for %v: %v", dir, fileinfo.Mode())
```
Comment (pweil-) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir/empty_dir.go]: Let's knock out https://github.com/pweil-/origin/commit/e09662d7281b832624f25738d986003cc2269b25#commitcomment-11634656 while we're in here so I won't have to do it for a follow up.

Comment (pmorie) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir/empty_dir.go]: Done

**Comment by pweil-** [comment]
LGTM - I'll remove the chmod commit from my PR and rebase on to this.

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**